File: api/UMS.Core/Entities/Applications.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace UMS.Core.Entities
{
    public class Applications
    {
        [Key]
        public int? id { get; set; }
        public int? type_id { get; set; }
        [ForeignKey("type_id")]
        public ApplicationType? type { get; set; }
        public int? zone_id { get; set; }
        [ForeignKey("zone_id")]
        public Zone? zone { get; set; }
        public string? observations {  get; set; }
        public int? approver_id {  get; set; }
        [ForeignKey(nameof(approver_id))]
        public User? approver { get; set; }
        public int? applicant_id { get; set; }
        [ForeignKey(nameof(applicant_id))]
        public User? applicant { get; set; }
        public DateTime created_at { get; set; }
        public DateTime? updated_at { get; set; }
        public bool? approved { get; set; }

    }
}

File: ums_app/lib/models/applications.dart
class Application {
  final int id;
  final int type_id;
  final int zone_id;
  final String observations;
  final int approver_id;
  final int applicant_id;
  final DateTime created_at;
  final DateTime updated_at;
  final bool approved;

  Application({
    required this.id,
    required this.type_id,
    required this.zone_id,
    required this.observations,
    required this.approver_id,
    required this.applicant_id,
    required this.created_at,
    required this.updated_at,
    required this.approved,
  });

  factory Application.fromJson(Map<String, dynamic> json) {
    return Application(
      id: json['id'],
      type_id: json['type_id'],
      zone_id: json['zone_id'],
      observations: json['observations'],
      approver_id: json['approver_id'],
      applicant_id: json['applicant_id'],
      created_at: DateTime.parse(json['created_at']),
      updated_at: DateTime.parse(json['updated_at']),
      approved: json['approved'],
    );
  }
}
